integration_tests/androidx_test/src/sharedTest/java/org/robolectric/integrationtests/axt/EspressoTest.java

```diff
@@ -223,4 +226,29 @@ public void clickButton_after_swipeUp() {
       activityScenario.onActivity(action -> assertThat(action.buttonClicked).isTrue());
     }
   }
+
+  @Test
+  public void focusButton_after_DpadDown() {
+    activityRule.getScenario().moveToState(Lifecycle.State.DESTROYED);
+    try (ActivityScenario<EspressoFocusableActivity> activityScenario =
+             ActivityScenario.launch(EspressoFocusableActivity.class)) {
+      onView(withId(R.id.scroll_view)).perform(ViewActions.pressKey(KeyEvent.KEYCODE_DPAD_DOWN));
+      onView(withId(R.id.scroll_view)).perform(ViewActions.pressKey(KeyEvent.KEYCODE_DPAD_DOWN));
+      activityScenario.onActivity(activity -> assertThat(activity.findViewById(R.id.button).isFocused()).isFalse());
+      activityScenario.onActivity(activity -> assertThat(activity.findViewById(R.id.button2).isFocused()).isTrue());
+    }
+  }
+
+  @GraphicsMode(GraphicsMode.Mode.NATIVE)
+  @Test
+  public void focusNativeButton_after_DpadDown() {
```

Comment:
**focusNativeButton_after_DpadDown**: This test is identical to `focusButton_after_DpadDown` but operates under `GraphicsMode.Mode.NATIVE`. It currently fails, indicating a bug when running in Native Graphics Mode. The focus does not shift to the second button as expected. This has been reported and is awaiting a fix.